src/parser/mod.rs

```diff
@@ -0,0 +1,85 @@
+use std::fmt;
+
+mod ascii;
+pub(crate) use ascii::{parse_ascii_response, parse_ascii_status, parse_version};
+
+use crate::ErrorKind;
+
+/// A value from memcached.
+#[derive(Clone, Debug, PartialEq)]
+pub struct Value {
+    /// The key.
+    pub key: Vec<u8>,
+    /// CAS identifier.
+    pub cas: Option<u64>,
+    /// Flags for this key.
+    ///
+    /// Defaults to 0.
+    pub flags: u32,
+    /// Data for this key.
+    pub data: Vec<u8>,
+}
+
+/// Status of a memcached operation.
+#[derive(Clone, Debug, PartialEq)]
+pub enum Status {
+    /// The value was stored.
+    Stored,
+    /// The value was not stored.
+    NotStored,
+    /// The key was deleted.
+    Deleted,
+    /// The key was touched.
+    Touched,
+    /// The key already exists.
+    Exists,
+    /// The key was not found.
+    NotFound,
+}
+
+/// Response to a memcached operation.
+#[derive(Clone, Debug, PartialEq)]
+pub enum Response {
+    /// The status of a given operation, which may or may not have succeeded.
+    Status(Status),
+    /// Data response, which is only returned for reads.
+    Data(Option<Vec<Value>>),
```

Comment:
I would probably not wrap this in `Option`, an empty vector is really the same as None in this case.